src/sql/magic.py

```diff
@@ -296,7 +295,9 @@ def _execute(self, payload, line, cell, local_ns):
             creator=args.creator,
             alias=args.alias,
         )
-        payload["connection_info"] = conn._get_curr_connection_info()
+        payload[
+            "connection_info"
+        ] = sql.connection.Connection._get_curr_connection_info()
```

Comment:
Why initialize two connections?

Reply:
Like maybe we can to a check and initialize a single one.

Also seems like this block with `payload` is repetitive between `plot.py` and `magics.py`

Reply:
Since this is the `payload["connection_info"]` here is for collecting `%sql [QUERY]`

However, in the `%sqlplot`, we still need to collect this info again since those are two command entry points